### PR TITLE
SLING-8374 Missing out exceptions in catch clause in DistributionSubs…

### PR DESCRIPTION
**Pull request by akrainiouk**
…criber class

### CODE REVIEW

File: src/main/java/org/apache/sling/distribution/journal/impl/subscriber/DistributionSubscriber.java [hunk 1]
```diff
@@ -552,8 +553,12 @@ private void importPackage(PackageMessage pkgMsg, long offset, long createdTime)
 
             Event event = DistributionEvent.eventImporterImported(pkgMsg, subAgentName);
             eventAdmin.postEvent(event);
-        } catch (Exception e) {
+        } catch (Throwable e) {
             distributionMetricsService.getFailedPackageImports().mark();
+            // rethrow fatal exceptions
+            if (e instanceof VirtualMachineError) {
```
Comment (tmaret) [inline src/main/java/org/apache/sling/distribution/journal/impl/subscriber/DistributionSubscriber.java]: We should re-throw all {{java.lang.Error}}

File: src/main/java/org/apache/sling/distribution/journal/impl/subscriber/DistributionSubscriber.java [hunk 2]
```diff
@@ -660,7 +665,7 @@ private void timedSendStoredStatus() {
             try {
                 sendStatusMessage(status);
                 sent = true;
-            } catch (Exception e) {
```
Comment (tmaret) [inline src/main/java/org/apache/sling/distribution/journal/impl/subscriber/DistributionSubscriber.java]: `MessagingException` extends from `Exception` and thus is already catch here. We want to catch all `Exception` exceptions here.

File: src/main/java/org/apache/sling/distribution/journal/impl/subscriber/DistributionSubscriber.java [hunk 3]
```diff
@@ -677,7 +682,7 @@ private void markStatusSent() {
         }
     }
 
-    private void sendStatusMessage(ValueMap status) {
+    private void sendStatusMessage(ValueMap status) throws MessagingException {
```
Comment (tmaret) [inline src/main/java/org/apache/sling/distribution/journal/impl/subscriber/DistributionSubscriber.java]: `MessagingException` is not a checked exception. By convention we don't declare runtime exceptions in method signatures.

File: src/main/java/org/apache/sling/distribution/journal/impl/subscriber/DistributionSubscriber.java (hunk 2 above)
Comment (akrainiouk) [reply src/main/java/org/apache/sling/distribution/journal/impl/subscriber/DistributionSubscriber.java]: Oops! My bad! Ok, in that case we don't need to touch this part.

File: src/main/java/org/apache/sling/distribution/journal/impl/subscriber/DistributionSubscriber.java (hunk 3 above)
Comment (akrainiouk) [reply src/main/java/org/apache/sling/distribution/journal/impl/subscriber/DistributionSubscriber.java]: OK

File: src/main/java/org/apache/sling/distribution/journal/impl/subscriber/DistributionSubscriber.java (hunk 1 above)
Comment (akrainiouk) [reply src/main/java/org/apache/sling/distribution/journal/impl/subscriber/DistributionSubscriber.java]: OK

**Pull request closed without merging**